crates/cli/src/main.rs

```diff
@@ -21,7 +21,14 @@ fn main() -> Result<()> {
     match args.command {
         Command::EmitProvider(opts) => emit_provider(&opts),
         Command::Compile(opts) => {
-            let js = JS::from_file(&opts.input)?;
+            let js = match opts.input.to_str() {
+                Some("-") => {
```

Comment:
On the technical side of this change, any particular reason why you decided chose hyphens over:

- Declaring the `input` and `output` options as optional via `Option<T>`
- Default to `stdin` / `stdout` accordingly if either or both are not specified

Reply:
For stdin input, I would argue that it is common practice to use a hyphen as the argument to indicate this explicitely. For the output, I didn't want to break the current default behavior (i.e. using `index.wasm`).

I see the point though, that especially the combination of both (`-o - -` / `- -o -`) may look a bit weird. In the end, I wanted to primarily make a proposal for the functionality – the implementation details can be discussed.

As this will anyway be implemented as part of the new `build` command, I would default the output to stdout as suggested by you.